.github/workflows/test.yaml

```diff
@@ -32,6 +32,29 @@ jobs:
       - name: Run static checks
         run: tox -e static
 
+  lint-commits: 
+    runs-on: ubuntu-latest
+    needs: static  
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-depth: 0  
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+      - name: Install gitlint
+        run: pip install gitlint>=0.20.0 
+      - name: Lint commits
+        run: |
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+            BASE_SHA=$(git merge-base HEAD origin/${{ github.base_ref }})
+            gitlint --commits "$BASE_SHA"..HEAD  # NEW: Lint only new commits in PR
+          else
+            gitlint --commits HEAD~10..HEAD  # NEW: Lint last 10 commits on push 
```

Comment:
If it isn't a pull request, it would probably be safest to assume that the `base_ref` would have been the repository's default branch (`github.event.repository.default_branch`)